node/src/action.rs

```diff
@@ -53,7 +55,7 @@ impl redux::EnablingCondition<crate::State> for CheckTimeoutsAction {}
 
 #[cfg(feature = "replay")]
 impl redux::EnablingCondition<crate::State> for Action {
-    fn is_enabled(&self, _: &crate::State, _time: redux::Timestamp) -> bool {
-        true
+    fn is_enabled(&self, state: &crate::State, _time: redux::Timestamp) -> bool {
+        !matches!(self, Action::InitializeP2p { .. }) || state.p2p.ready().is_none()
```

Comment:
@binier `InitializeP2p` shouldn't be triggered twice. How can I express that?

Reply:
@akoptelov in this case you need to define a type for this action like in the TODO comment, so that that action is dispatched (and `is_enabled` for that type is checked) instead of `Action::InitializeP2p` directly.

Note that queueing `is_enabled` for `Action` is always called, never directly for the subtype (simplifes a lot of stuff, and having `Action::is_enabled` behave differently from the nested types is also confusing). But somehow during replay it should still return `true` I assume (this here reminded me that I still need to solve that)

Reply:
This action enables p2p, not called in it.

@binier if I just wrap it into another enum, would it help?

Reply:
Ok, got that now, with EC for Action. Moved to P2pAction, as you suggested.